core/src/main/java/hudson/PluginManager.java

```diff
@@ -506,6 +508,15 @@ protected boolean filter(Method e) {
         }
 
         LOGGER.info("Plugin " + p.getShortName()+":"+p.getVersion() + " dynamically installed");
+
+        // If this plugin is a UIThemeContributor, tell the UIThemesProcessor about it.
+        // TODO: I don't see a plugin uninstall sequence ?
+        if (isUIThemesPluginInstalled()) {
+            Plugin pluginInstance = p.getPlugin();
+            if (pluginInstance instanceof UIThemeContributor) {
+                UIThemesProcessor.getInstance().addContributor((UIThemeContributor) pluginInstance);
+            }
+        }
```

Comment:
Yeah that would be nicer if it existed. I don't think it exists though, right? Might be overkill introducing it yet though.